lldb/include/lldb/Symbol/TypeSystem.h

```diff
@@ -146,6 +133,9 @@ class TypeSystem : public PluginInterface,
   virtual CompilerDeclContext
   GetCompilerDeclContextForType(const CompilerType &type);
 
+  virtual std::vector<lldb_private::CompilerContext>
+  DeclContextGetCompilerContext(void *opaque_decl_ctx);
```

Comment:
I would rather not put any limitations on TypeSystem plug-ins in case they can be more efficient if they can put off parsing information about a type until accessors are called on the `DeclContext` objects.